javascript/src/main/pages/AnalyzeMessageData/AnalyzeReactions.js

```diff
@@ -1,7 +1,46 @@
-import React from "react";
+import React, {useState} from "react";
+import { useAuth0 } from "@auth0/auth0-react";
+import useSWR from "swr";
+import { fetchWithToken } from "main/utils/fetch";
+import MessageTableReaction from "main/components/ChannelMessageReaction/MessageTableReaction"
+import { Form, Button } from "react-bootstrap";
+
 
 const AnalyzeReactions = () => {
-    return <h1>Analyze Reactions</h1>
+    const { getAccessTokenSilently: getToken } = useAuth0();
+    const [ searchReaction, setSearchReaction ] = useState('');
+    const [ searchResults, setSearchResults ] = useState([]);
+    
+    const handleSearchReactionOnChange = (event) => {
+        setSearchReaction(event.target.value);
+    };
+    
+    const handleSearchReactionOnSubmit = () => {
+        let url = (searchReaction=="+1") ? `/api/members/messages/reactionsearch?searchReaction=%2B1` : `/api/members/messages/reactionsearch?searchReaction=${searchReaction}`;
```

Comment:
This line of code is URLEncoding +1 as a special case.

I wonder if it would be better to URLEncode everything using https://www.w3schools.com/jsref/jsref_encodeuri.asp

e.g. 

```
         let url = `/api/members/messages/reactionsearch?searchReaction=${URIEncode(searchReaction)}`;
```